include/boost/geometry/extensions/random/strategies/agnostic/uniform_convex_fan.hpp

```diff
@@ -0,0 +1,95 @@
+// Boost.Geometry (aka GGL, Generic Geometry Library)
+
+// Copyright (c) 2019 Tinko Bartels, Berlin, Germany.
+
+// Use, modification and distribution is subject to the Boost Software License,
+// Version 1.0. (See accompanying file LICENSE_1_0.txt or copy at
+// http://www.boost.org/LICENSE_1_0.txt)
+
+#ifndef BOOST_GEOMETRY_EXTENSIONS_RANDOM_STRATEGIES_AGNOSTIC_UNIFORM_CONVEX_FAN_HPP
+#define BOOST_GEOMETRY_EXTENSIONS_RANDOM_STRATEGIES_AGNOSTIC_UNIFORM_CONVEX_FAN_HPP
+
+#include <random>
+#include <vector>
+#include <cstdlib>
+#include <cmath>
+
+#include <boost/geometry/algorithms/equals.hpp>
+#include <boost/geometry/algorithms/transform.hpp>
+#include <boost/geometry/algorithms/within.hpp>
+
+namespace boost { namespace geometry
+{
+
+namespace strategy { namespace uniform_point_distribution {
+
+// The following strategy is suitable for convex rings and polygons with
+// non-empty interior.
+template
+<
+    typename Point,
+    typename DomainGeometry,
+    typename TriangleStrategy,
+    typename SideStrategy   //Actually, we need a triangle area strategy here.
+>
+struct uniform_convex_fan
+{
+private:
+    std::vector<double> accumulated_areas;
+    // It is hard to see a reason not to use double here. If a triangles
+    // relative size is smaller than doubles epsilon, it is too unlikely to
+    // realistically occur in a random sample anyway.
+public:
+    uniform_convex_fan(DomainGeometry const& g)
+	{
+        accumulated_areas.push_back(0);
+        for (int i = 2 ; i < g.size() ; ++i) {
+            accumulated_areas.push_back(
+                accumulated_areas.back() +
+                std::abs(SideStrategy::template side_value<double, double>(
+                         *g.begin(),
+                         *(g.begin() + i - 1),
+                         *(g.begin() + i))));
+        }
+    }
+    bool equals(DomainGeometry const& l_domain,
```

Comment:
The notion of equality as defined by the C++ named requirement "RandomNumberDistribution" on which I based my design of random point distributions considers two distribution instances A, B equal if their parameters are equal and they will produce the same output given the same generators, i.e. from A == B and g1 == g2 it needs to follow that A(g1) == A(g2).

Because of the way I generate samples for convex rings, two distributions will only produce the same output if the convex rings start from the same point.

Reply:
OK, I see, I remember also our discussions on this topic. Please add a comment or document this for future reference